src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java

```diff
@@ -430,18 +446,29 @@ private static void appendManifestHeader(StringBuilder manifest) {
                 errors.println();
             }
         }
-        return contents.contents;
+        return contentsContainer.contents;
     }
 
     private static class ContentContainer extends Container {
         private final List<Content> contents = new ArrayList<>();
         private final Set<String> names = new TreeSet<>();
 
+        //The filter to return the names filtered
+        private final Optional<ContentFilter> maybeFilter;
+
+        /**
+         * We need the filter to be able to filter the contents written to the manifest
+         * @param maybeFilter filter to use when writing the name of the contents
+         */
+        public ContentContainer(Optional<ContentFilter> maybeFilter) {
```

Comment:
The new constructor (private class) to filter the names every time a content is added to this container. Weird architecture BTW